artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -611,13 +614,24 @@ public void setProperties(String fileUrltoBrokerProperties) {
       propertiesFileUrl = fileUrltoBrokerProperties;
    }
 
+   @Override
+   public String getStatus() {
+      return serverStatus.asJson();
+   }
+
+   @Override
+   public void updateStatus(String component, String statusJson) {
+      serverStatus.update(component, statusJson);
+   }
+
    private void internalStart() throws Exception {
       if (state != SERVER_STATE.STOPPED) {
          logger.debug("Server already started!");
          return;
       }
 
       configuration.parseProperties(propertiesFileUrl);
+      updateStatus("configuration", configuration.getStatus());
```

Comment:
status names should probably have a constant somewhere to help consistency, knowing what exists, easily finding related tests..etc

Reply:
maybe, not sure where to put them... they mostly should be a direct map from module/component name to status attribute, and within those objects - attribute are some subset of useful named values, using their names.
the "name" could be inferred really.

Reply:
It could just be in the impl classes if its tied to the impl+tests, or a specific class for more general ones. Anywhere really, that helps tie the related bits together rather than expecting people will properly follow/associate literals later. This class has 2 related updates for "configuration" but thats not so obvious..whereas following a shared ref to the name would make it much more so.

Reply:
agree, thanks.